zipline/utils/events.py

```diff
@@ -539,7 +539,7 @@ def get_first_trading_day_of_month(self, dt):
         self.month = dt.month
 
         dt = dt.replace(day=1)
-        first_day = (dt if self.cal.is_open_on_day(dt)
+        first_day = (normalize_date(dt) if self.cal.is_open_on_day(dt)
```

Comment:
One thought here might be that we need a method with a signature like:

```
def find_nearest_session_date(self, dt, search_direction):
    """
    Find the nearest session date to dt.

    If dt is in a trading session, return that session's label.
    Otherwise, if search_direction == 'forward', find the label of the first session starting after dt;
    if search_direction == 'backward', find the label of the latest session ending before dt.
    """
```
    
```